mathesar_ui/src/stores/table-data/tabularData.ts

```diff
@@ -61,8 +61,9 @@ export class TabularData extends EventHandler {
     );
 
     this.metaParametersUnsubscriber = this.meta.metaParameters.subscribe(() => {
-      this.dispatch('paramsUpdated', this.parameterize());
+      void this.dispatch('paramsUpdated', this.parameterize());
     });
+    this.columnsDataStore.on('columnRenamed', () => this.refresh());
```

Comment:
We also need to update the meta object. Perhaps have a method within meta to handle renames and call it from here.

This would not be essential after column name changes to id in our APIs, but until then I'd like the existing features not to break.

Reply:
> I'd like the existing features not to break.

Given that renaming a column is not an existing feature, what existing features will break by merging this PR?

Even if something small _does_ break, I'd rather not do the extra work now, given that we know we'll need to undo it soon.

Reply:
I'm sorry, there aren't breaking features, I meant to resolve the delete bug that is already present. I see an issue for it here: https://github.com/centerofci/mathesar/issues/987. 

I'll resolve this and we can handle the issue there.